tnt-service/src/main/java/org/dcsa/tnt/service/mapping/transferobject/LocationMapper.java

```diff
@@ -0,0 +1,60 @@
+package org.dcsa.tnt.service.mapping.transferobject;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.dcsa.tnt.service.domain.Facility;
+import org.dcsa.tnt.service.domain.Location;
+import org.dcsa.tnt.transferobjects.AddressLocationTO;
+import org.dcsa.tnt.transferobjects.FacilityLocationTO;
+import org.dcsa.tnt.transferobjects.LocationTO;
+import org.dcsa.tnt.transferobjects.UNLocationLocationTO;
+import org.dcsa.tnt.transferobjects.enums.FacilityCodeListProvider;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@AllArgsConstructor
+public class LocationMapper {
+  private final AddressMapper addressMapper;
+
+  public LocationTO toDTO(Location location) {
+    if (location == null) {
+      return null;
+    }
+
+    if (location.address() != null) {
+      return AddressLocationTO.builder()
+        .locationName(location.locationName())
+        .address(addressMapper.toDomain(location.address()))
+        .build();
+    } else if (location.facility() != null) {
+      Facility facility = location.facility();
+      String facilityCode;
+      FacilityCodeListProvider facilityCodeListProvider;
+      if (facility.facilitySMDGCode() != null) {
+        facilityCode = facility.facilitySMDGCode();
+        facilityCodeListProvider = FacilityCodeListProvider.SMDG;
+      } else if (facility.facilityBICCode() != null) {
+        facilityCode = facility.facilityBICCode();
+        facilityCodeListProvider = FacilityCodeListProvider.BIC;
+      } else {
+        log.error("Facility {} has neither SMDG code nor BIC code", facility.id());
+        return null;
```

Comment:
I think this warrants an exception (`IllegalArgumentException`)

Reply:
Fixed